crates/ecs/src/lib.rs

```diff
@@ -806,4 +1138,391 @@ mod tests {
 
         assert_eq!(expected_accesses, component_accesses)
     }
+
+    // Archetype tests:
+
+    #[test]
+    fn archetype_can_store_components_of_entities_it_stores() {
+        let mut archetype = Archetype::default();
+
+        let entity_1_id = 0;
+        let entity_2_id = 10;
+
+        // 1. Archetype stores the components of entities with id 0 and id 10
+        archetype.store_entity(entity_1_id);
+        archetype.store_entity(entity_2_id);
+
+        // 2. Create component vectors for types u32 and u64
+        archetype.add_component_vec::<u32>();
+        archetype.add_component_vec::<u64>();
+
+        // 3. Add components for entity id 0
+        archetype.add_component::<u32>(entity_1_id, 21);
+        archetype.add_component::<u64>(entity_1_id, 212);
+
+        // 4. Add components for entity id 1
+        archetype.add_component::<u32>(entity_2_id, 35);
+        archetype.add_component::<u64>(entity_2_id, 123);
+
+        let result_u32 = archetype.borrow_component_vec::<u32>().unwrap();
+        let result_u64 = archetype.borrow_component_vec::<u64>().unwrap();
+
+        // entity 1 should have index 0, as it was added first
+        assert_eq!(result_u32.get(0).unwrap().unwrap(), 21);
+        assert_eq!(result_u64.get(0).unwrap().unwrap(), 212);
+
+        // entity 2 should have index 1 as it was added second
+        assert_eq!(result_u32.get(1).unwrap().unwrap(), 35);
+        assert_eq!(result_u64.get(1).unwrap().unwrap(), 123);
+    }
+
+    #[test]
+    fn components_are_stored_on_correct_index_independent_of_insert_order() {
+        let mut archetype = Archetype::default();
+
+        let entity_1_id = 0;
+        let entity_2_id = 10;
+        let entity_3_id = 5;
+
+        let entity_1_idx = 0;
+        let entity_2_idx = 1;
+        let entity_3_idx = 2;
```

Comment:
nit:
I assume these are indices and are not meant to be read as "identifier x"? If so, you could probably afford the extra characters to just write out `entity_n_index`